### PR TITLE
change keepalived settings, add envoy tcp idle timeout, add envoy tcp idle timeout metric

### PR DESCRIPTION
**Pull request by dergeberl**
change keepalived settings:
- advert_int to 4 (because of failed swaps while openstack migrations)
- default state to BACKUP

add envoy tcp/udp idle timeout:
- make it configurable 

add envoy tcp idle timeout metric:
- add metric to lbm for number of tcp idle timeouts 
- add tests for metric generation

### CODE REVIEW

File: internal/helper/yawollet.go [hunk 1]
```diff
@@ -301,6 +301,7 @@ func createEnvoyTCPListener(
 	listenPort, err := anypb.New(&envoytcp.TcpProxy{
 		StatPrefix:       "envoytcp",
 		ClusterSpecifier: &envoytcp.TcpProxy_Cluster{Cluster: fmt.Sprintf("%v-%v", port.Protocol, port.Port)},
+		IdleTimeout:      &duration.Duration{Seconds: 300}, // default is one hour, prevent idle connection in cases of openstack server migration
```
Comment (nschad) [inline internal/helper/yawollet.go:304]:
```suggestion
		IdleTimeout:      &duration.Duration{Seconds: 60}, // default is one hour, prevent idle connection in cases of openstack server migration
```

can be even lower. AWS uses 60s.

**Comment by dergeberl** [comment]
> Updates to metrics should be reflected in the respective tables of the [components overview](https://github.com/stackitcloud/yawol/blob/main/docs/components.md).

Good point I added all envoy metrics to the list.

File: internal/helper/yawollet.go (hunk 1 above)
Comment (dergeberl) [reply internal/helper/yawollet.go:304]: I researched more and see that this is a complicated topic. Some applications can have problems with low settings. I decided to make it configurable and leave the default as it is, to not break something.

**Review by breuerfelix — APPROVED** [review]

File: api/v1beta1/loadbalancer_types.go [hunk 2]
```diff
@@ -104,6 +108,16 @@ type LoadBalancerOptions struct {
 	// If empty it is enabled for all ports. Only has an affect if TCPProxyProtocol is enabled.
 	// +optional
 	TCPProxyProtocolPortsFilter []int32 `json:"tcpProxyProtocolPortFilter,omitempty"`
+	// TCPIdleTimeout sets TCP idle Timeout for all TCP connections from this LoadBalancer.
+	// Value is in Seconds. With 0 you disable the idle timeout, be careful this can lead to side effects.
+	// Default is 3600s (1 hour).
+	// +optional
+	TCPIdleTimeout *int `json:"tcpIdleTimeout,omitempty"`
```
Comment (nightlyone) [inline api/v1beta1/loadbalancer_types.go:115]: I wonder why you don't support durations here? [k8s](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Duration) supports them, [Go](https://pkg.go.dev/time#Duration) supports them, [protobuf](https://pkg.go.dev/google.golang.org/protobuf/types/known/durationpb) supports them too.

Durations make it clearer what unit are your talking about without looking at the documentation and allow you to go sub-second when your timings get tighter.

Comment (dergeberl) [reply api/v1beta1/loadbalancer_types.go:115]: Good point, changed in #73

**Pull request merged**